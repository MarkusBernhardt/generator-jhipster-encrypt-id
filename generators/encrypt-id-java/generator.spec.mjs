import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'scmb-encrypt-id-java';
const SUB_GENERATOR_NAMESPACE = `jhipster-scmb-encrypt-id:${SUB_GENERATOR}`;

describe('SubGenerator scmb-encrypt-id-java of scmb-encrypt-id JHipster blueprint', () => {
  describe('run', () => {
    beforeAll(async function () {
      await helpers
        .run(SUB_GENERATOR_NAMESPACE)
        .withJHipsterConfig({}, [
          {
            name: 'Entities',
            enableAudit: true,
            fields: [
              {
                fieldName: 'name',
                fieldType: 'String',
              },
            ],
          },
        ])
        .withOptions({
          creationTimestamp: '2024-02-01',
          ignoreNeedlesError: true,
        })
        .withJHipsterLookup()
        .withParentBlueprintLookup();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
