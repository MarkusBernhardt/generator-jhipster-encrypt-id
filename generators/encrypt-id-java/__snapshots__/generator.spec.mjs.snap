// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator encrypt-id-java of encrypt-id JHipster blueprint > run > should succeed 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
}
`;
